File: 27.java

class Solution1 {
    public static int removeElement(int[] nums, int val) {
        int k = 0;

        for (int loop = 0; loop < nums.length; loop++){
            if (nums[loop] == val){
                k += 1;
            }
            if (nums[loop] != val){
                nums[loop - k] = nums[loop];
            }
        }
        return nums.length - k;
    }
}

File: Python/27.py
from typing import List


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        counter = 0
        left = 0
        while left < len(nums) - counter:
            if nums[left] == val:
                nums[left] = nums[len(nums) - 1 - counter]
                counter += 1
            else:
                left += 1
        return len(nums) - counter
